skimage/filters/_sparse_cy.pyx

```diff
@@ -0,0 +1,24 @@
+import cython
+cimport numpy as np
+from .._shared.fused_numerics cimport np_floats
+
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+def _correlate_sparse_offsets(np_floats[:] input, Py_ssize_t[:] indices, 
+                              Py_ssize_t[:] offsets, np_floats[:] values, 
+                              np_floats[:] output):
+    cdef Py_ssize_t i
+    cdef Py_ssize_t indices_len = indices.shape[0]
+    cdef Py_ssize_t value_len = values.shape[0]
+    cdef Py_ssize_t off
+    cdef np_floats val
+    cdef Py_ssize_t vindex
+    
+    for vindex in range(value_len):
+        off = offsets[vindex]
+        val = values[vindex]
```

Comment:
```suggestion
    for off, val in zip(offsets, values):
```

Reply:
@rfezzani good points! One should check that the above suggestion does not harm the benchmarks.

@zeroth you can release the gil by wrapping the loop code in a

```python
with nogil:
    do_something...
```

block

Reply:
Hi @rfezzani and @jni,
Quick update,
if I just update the current code to  `for off, val in zip(offsets, values):`  there no performance hit.
But, if I add 
```
with nogil:
        for off, val in zip(offsets, values):
```
Then there is a compile error 
```
with nogil:
        for off, val in zip(offsets, values):
                                            ^
skimage\filters\_sparse_cy.pyx:23:37: Converting to Python object not allowed without gil
```
I think making the thread-safe is priority hence I have just added `with nogil`. 
Let me know your thoughts.